compiler/noirc_frontend/src/hir/mod.rs

```diff
@@ -37,15 +41,59 @@ pub enum FunctionNameMatch<'a> {
 
 impl Context {
     pub fn new(file_manager: FileManager, crate_graph: CrateGraph) -> Context {
-        Context {
+        let mut context = Context {
             def_interner: NodeInterner::default(),
             def_maps: BTreeMap::new(),
             visited_files: BTreeMap::new(),
             crate_graph,
             file_manager,
+            file_to_ast_cache: HashMap::default(),
+        };
+
+        context.populate_cache_with_file_manager();
+
+        context
+    }
+
+    // TODO: This method should live at a higher level
+    fn populate_cache_with_file_manager(&mut self) {
+        for file_id in self.file_manager.as_file_map().all_file_ids() {
+            let file_path = self.file_manager.path(*file_id);
+            let file_extension =
+                file_path.extension().expect("expected all file paths to have an extension");
+            // TODO: Another reason we may not want to have this method here
+            // TODO: is the fact that we want to not have the compiler worry
+            // TODO about the nr file extension
+            if file_extension != "nr" {
+                continue;
+            }
+            self.file_to_ast_cache.insert(*file_id, parse_file(&self.file_manager, *file_id));
         }
     }
 
+    // TODO: Do not merge with this method!
+    pub fn repopulate_cache(&mut self) {
+        for file_id in self.file_manager.as_file_map().all_file_ids() {
+            let file_path = self.file_manager.path(*file_id);
+            let file_extension =
+                file_path.extension().expect("expected all file paths to have an extension");
+            // TODO: Another reason we may not want to have this method here
+            // TODO: is the fact that we want to not have the compiler worry
+            // TODO about the nr file extension
+            if file_extension != "nr" {
+                continue;
+            }
+            self.file_to_ast_cache.insert(*file_id, parse_file(&self.file_manager, *file_id));
+        }
+    }
```

Comment:
Here is the hack, I added to repopulate the cache once I know we have added the stdlib to the file manager

Reply:
I just copied the above method and made this one public